File: forensics-btmgen/src/main/java/de/burger/forensics/plugin/engine/JavaScanner.java
package de.burger.forensics.plugin.engine;

import java.util.List;

/**
 * Scans a Java source text and returns Byteman rules for methods/branches found.
 * The emitter uses METHOD(name(..)) and escapes/truncates strings as needed.
 */
public interface JavaScanner {
    List<String> scan(
            String text,
            String helperFqn,
            String packagePrefix,
            boolean includeEntryExit,
            int maxStringLength);
}

File: forensics-btmgen/src/main/kotlin/de/burger/forensics/plugin/engine/JavaScanner.kt
package de.burger.forensics.plugin.engine

interface JavaScanner {
    /**
     * Scans a Java source text and returns Byteman rules for methods/branches found.
     * The emitter uses METHOD(name(..)) and escapes/truncates strings as needed.
     */
    fun scan(text: String, helperFqn: String, packagePrefix: String?, includeEntryExit: Boolean, maxStringLength: Int = 0): List<String>
}